services/catalog/src/simcore_service_catalog/db/repositories/_services_sql.py

```diff
@@ -320,66 +249,91 @@ def list_latest_services_with_history_stmt(
     )
 
 
-def list_services_with_history_stmt(
+def get_service_stmt(
     *,
     product_name: ProductName,
     user_id: UserID,
     access_rights: sa.sql.ClauseElement,
-    limit: int | None,
-    offset: int | None,
+    service_key: ServiceKey,
+    service_version: ServiceVersion,
 ):
-    #
-    # Common Table Expression (CTE) to select distinct service names with pagination.
-    # This allows limiting the subquery to the required pagination instead of paginating at the last query.
-    # SEE https://learnsql.com/blog/cte-with-examples/
-    #
-    cte = (
-        sa.select(services_meta_data.c.key)
-        .distinct()
+    owner_subquery = (
+        sa.select(users.c.email)
+        .select_from(user_to_groups.join(users, user_to_groups.c.uid == users.c.id))
+        .where(user_to_groups.c.gid == services_meta_data.c.owner)
+        .limit(1)
+        .scalar_subquery()
+    )
+
+    return (
+        sa.select(
+            services_meta_data.c.key,
+            services_meta_data.c.version,
+            # display
+            services_meta_data.c.name,
+            services_meta_data.c.description,
+            services_meta_data.c.thumbnail,
+            # ownership
+            owner_subquery.label("owner_email"),
+            # tags
+            services_meta_data.c.classifiers,
+            services_meta_data.c.quality,
+            # lifetime
+            services_meta_data.c.created,
+            services_meta_data.c.modified,
+            services_meta_data.c.deprecated,
+            # w/o releases history!
+        )
         .select_from(
             services_meta_data.join(
                 services_access_rights,
                 (services_meta_data.c.key == services_access_rights.c.key)
-                & (services_meta_data.c.version == services_access_rights.c.version)
-                & (services_access_rights.c.product_name == product_name),
+                & (services_meta_data.c.version == services_access_rights.c.version),
             ).join(
-                user_to_groups,
-                (user_to_groups.c.gid == services_access_rights.c.gid)
-                & (user_to_groups.c.uid == user_id),
+                user_to_groups, (user_to_groups.c.gid == services_access_rights.c.gid)
             )
         )
-        .where(access_rights)
-        .order_by(services_meta_data.c.key)  # NOTE: add here the order
-        .limit(limit)
-        .offset(offset)
-        .cte("paginated_services")
+        .where(
+            (services_meta_data.c.key == service_key)
+            & (services_meta_data.c.version == service_version)
+            & (user_to_groups.c.uid == user_id)
+            & (services_access_rights.c.product_name == product_name)
+            & access_rights
+        )
     )
```

Comment:
Out of curiosity, did you test the performance of these queries on a db which holds a lot of data (e.g. inhouse master)?